### PR TITLE
[API Compatibility]fix the loss doc

### PR DESCRIPTION
**Pull request by zty-king**
17种存在size_average与reduce的loss，添加size_average+reduce->reduction的转写示例

### CODE REVIEW

**Comment by zty-king** [comment]
@zhwesky2010 您有空了看看这个格式是否可行，这里CI报错但是报错文档打不开，实在不知道为什么有问题

File: docs/guides/model_convert/convert_from_pytorch/api_difference/torch_more_args/torch.nn.BCELoss.md [hunk 1]
```diff
@@ -25,65 +25,35 @@ PyTorch 相比 Paddle 支持更多其他参数，具体如下：
...
-
-
-#### size_average
-size_average 为 True
+#### reduction 为 sum
 ```python
 # PyTorch 写法
-torch.nn.BCELoss(size_average=True)
+torch.nn.BCELoss(weight=w, size_average=False, reduce=True)
+torch.nn.BCELoss(weight=w, size_average=False)
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/torch_more_args/torch.nn.BCELoss.md:32]: 还有一种写法
torch.nn.BCELoss(weight=w, reduction='sum)

File: docs/guides/model_convert/convert_from_pytorch/api_difference/torch_more_args/torch.nn.BCELoss.md [hunk 2]
```diff
@@ -25,65 +25,35 @@ PyTorch 相比 Paddle 支持更多其他参数，具体如下：
...
 
 # Paddle 写法
-paddle.nn.BCELoss(reduction='mean')
+paddle.nn.BCELoss(weight=w, reduction='sum')
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/torch_more_args/torch.nn.BCELoss.md:35]: 注释一下：
```
## 以上写法都统一对应到如下写法
```

File: docs/guides/model_convert/convert_from_pytorch/api_difference/torch_more_args/torch.nn.BCELoss.md [hunk 3]
```diff
@@ -25,65 +25,35 @@ PyTorch 相比 Paddle 支持更多其他参数，具体如下：
 | reduction  | reduction            | 表示应用于输出结果的计算方式。  |
 
 ### 转写示例
-
-
-#### size_average
-size_average 为 True
+#### reduction 为 sum
```
Comment (zhwesky2010) [inline docs/guides/model_convert/convert_from_pytorch/api_difference/torch_more_args/torch.nn.BCELoss.md]: 一共有34种用法，还有torch.nn.functional.* 下面有17个

File: docs/guides/model_convert/convert_from_pytorch/api_difference/torch_more_args/torch.nn.BCELoss.md (hunk 1 above)
Comment (zty-king) [reply docs/guides/model_convert/convert_from_pytorch/api_difference/torch_more_args/torch.nn.BCELoss.md:32]: 这里我看不是之前都只给size_average+reduce到reduction的转写示例嘛，reduction到reduction是一一对应的关系，还要写出来吗

File: docs/guides/model_convert/convert_from_pytorch/api_difference/torch_more_args/torch.nn.BCELoss.md (hunk 3 above)
Comment (zty-king) [reply docs/guides/model_convert/convert_from_pytorch/api_difference/torch_more_args/torch.nn.BCELoss.md]: 这个路径下的loss，我看paddle都没支持size_average，reduce参数，要支持一下吗

Comment (zhwesky2010) [reply docs/guides/model_convert/convert_from_pytorch/api_difference/torch_more_args/torch.nn.BCELoss.md]: > 这个路径下的loss，我看paddle都没支持size_average，reduce参数，要支持一下吗

需要支持。同时之前的的legacy_reduction_decorator也需要在loss.py的functional API里都支持下。

**Comment by zty-king** [comment]
好，那我先对这些api做支持，再来改文档吧

---- Replied Message ----
| From | ***@***.***> |
| Date | 11/18/2025 16:17 |
| To | PaddlePaddle/docs ***@***.***> |
| Cc | Tianyu ***@***.***>,
Author ***@***.***> |
| Subject | Re: [PaddlePaddle/docs] [API Compatibility]fix the loss doc (PR #7632) |

@zhwesky2010 commented on this pull request.

In docs/guides/model_convert/convert_from_pytorch/api_difference/torch_more_args/torch.nn.BCELoss.md:

> @@ -25,65 +25,35 @@ PyTorch 相比 Paddle 支持更多其他参数，具体如下：
 | reduction  | reduction            | 表示应用于输出结果的计算方式。  |
 
 ### 转写示例
-
-
-#### size_average
-size_average 为 True
+#### reduction 为 sum


这个路径下的loss，我看paddle都没支持size_average，reduce参数，要支持一下吗

需要支持。paddle都不支持size_average、reduce参数吧

File: docs/guides/model_convert/convert_from_pytorch/api_difference/torch_more_args/torch.nn.BCELoss.md (hunk 1 above)
Comment (zhwesky2010) [reply docs/guides/model_convert/convert_from_pytorch/api_difference/torch_more_args/torch.nn.BCELoss.md:32]: 这个也可以不写，是一一对应的

File: docs/guides/model_convert/convert_from_pytorch/api_difference/torch_more_args/torch.nn.BCELoss.md (hunk 3 above)
Comment (zty-king) [reply docs/guides/model_convert/convert_from_pytorch/api_difference/torch_more_args/torch.nn.BCELoss.md]: 由于loss.py下都是函数（之前的17个loss是类），所以需要在原来的legacy_reduction_decorator里新增一个判断，因为原来算位置参数需要跳过self，如果是函数的话就不用跳过，会引入新的开销；或者就再写一个装饰器，不过大部分代码都可以复用，您看哪个好一点

Comment (zty-king) [reply docs/guides/model_convert/convert_from_pytorch/api_difference/torch_more_args/torch.nn.BCELoss.md]: > 由于loss.py下都是函数（之前的17个loss是类），所以需要在原来的legacy_reduction_decorator里新增一个判断，因为原来算位置参数需要跳过self，如果是函数的话就不用跳过，会引入新的开销；或者就再写一个装饰器，不过大部分代码都可以复用，您看哪个好一点

@zhwesky2010 您有空了看一下这个哈

Comment (zty-king) [reply docs/guides/model_convert/convert_from_pytorch/api_difference/torch_more_args/torch.nn.BCELoss.md]: @zhwesky2010 您有空了看看这个评论哈

**Comment by zhwesky2010** [comment]
@zty-king 先把这个合入吧，修改的这些文档没有通过格式校验，看下CI报错

infoflow 2025-11-26 12-25-50

**Review by zhwesky2010 — APPROVED** [review]
LGTM
